staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -132,24 +131,39 @@ func NonSlidingUntilWithContext(ctx context.Context, f func(context.Context), pe
 // Close stopCh to stop. f may not be invoked if stop channel is already
 // closed. Pass NeverStop to if you don't want it stop.
 func JitterUntil(f func(), period time.Duration, jitterFactor float64, sliding bool, stopCh <-chan struct{}) {
-	BackoffUntil(f, NewJitteredBackoffManager(period, jitterFactor, &clock.RealClock{}), sliding, stopCh)
+	b := Backoff{Duration: period, Jitter: jitterFactor}
+	BackoffUntil(f, RealTimer, b.Step, sliding, stopCh)
 }
 
+var (
+	// RealTimer can be passed to methods that need a TimerFunc.
+	RealTimer = clock.RealClock{}.NewTimer
+)
+
+// TimerFunc allows the caller to inject a timer for testing. Most callers should use RealTimer.
```

Comment:
if that is the case should we even expose it?
or just make it private to override only on these tests?

Reply:
Timer also says https://pkg.go.dev/time#Timer

>  A Timer must be created with NewTimer or AfterFunc.

that may open another unsafe options for users , or can they create this `clock.Timer` without `NewTimer` ?

Reply:
Good point, I think TimerFunc can be safely removed now (delay func and Timer interface cover it).

Reply:
Updated in latest.